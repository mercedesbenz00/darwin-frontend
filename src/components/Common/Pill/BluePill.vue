<template>
  <pill-base class="blue-pill">
    <slot />
  </pill-base>
</template>

<script lang="ts">
import { defineComponent } from 'vue'

import PillBase from './PillBase.vue'

export default defineComponent({ name: 'BluePill', components: { PillBase } })
</script>

<style lang="scss" scoped>
@import "./pill";

.blue-pill {
  @include pill(
    rgba(160, 198, 234, 0.5),
    rgba(72, 130, 243, 0.7),
    rgba(0, 123, 237, 0.35)
  );
}
</style>
