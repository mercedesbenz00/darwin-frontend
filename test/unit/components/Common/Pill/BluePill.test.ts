import { createLocalVue, shallowMount } from '@vue/test-utils'

import BluePill from '@/components/Common/Pill/BluePill.vue'

const localVue = createLocalVue()

const slots = {
  default: 'A message'
}

it('matches snapshot', () => {
  const wrapper = shallowMount(BluePill, { localVue, slots })
  expect(wrapper).toMatchSnapshot()
})

it('renders slot', () => {
  const wrapper = shallowMount(BluePill, { localVue, slots })
  expect(wrapper.text()).toContain(slots.default)
})
