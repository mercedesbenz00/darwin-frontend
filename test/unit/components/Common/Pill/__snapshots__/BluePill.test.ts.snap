// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `<pill-base-stub class="blue-pill">A message</pill-base-stub>`;
